packages/util/src/bytes.ts

```diff
@@ -444,5 +444,66 @@ export const concatBytes = (...arrays: Uint8Array[]): Uint8Array => {
   return result
 }
 
+/**
+ * @notice Read a 32-bit little-endian integer from a Uint8Array
+ * @param {Uint8Array} bytes The input Uint8Array from which to read the 32-bit integer.
+ * @return {number} The 32-bit little-endian integer read from the input Uint8Arrays.
+ */
+export function readInt32LE(bytes: Uint8Array): number {
+  return (bytes[0] | (bytes[1] << 8) | (bytes[2] << 16) | (bytes[3] << 24)) >>> 0
+}
+
+/**
+ * @notice Read a 64-bit little-endian bigint from a Uint8Array
+ * @param {Uint8Array} bytes The input Uint8Array from which to read the 64-bit bigint.
+ * @return {bigint} The 64-bit little-endian bigint read from the input Uint8Arrays.
+ */
+export function readBigInt64LE(bytes: Uint8Array): bigint {
+  const lo = BigInt((bytes[0] | (bytes[1] << 8) | (bytes[2] << 16) | (bytes[3] << 24)) >>> 0)
+  const hi = BigInt((bytes[4] | (bytes[5] << 8) | (bytes[6] << 16) | (bytes[7] << 24)) >>> 0)
+
+  return (hi << BigInt(32)) | lo
+}
+
+/**
+ * @notice Write a 32-bit little-endian number to a Uint8Array.
+ * @param {number} number The number value to write to the Uint8Array.
+ * @return {Uint8Array} A Uint8Array of length 32 containing the 32-bit little-endian number.
+ */
+export function writeInt32LE(number: number): Uint8Array {
+  const bytes = new Uint8Array(32)
+
+  bytes[0] = number & 0xff
+  bytes[1] = (number >> 8) & 0xff
+  bytes[2] = (number >> 16) & 0xff
+  bytes[3] = (number >> 24) & 0xff
+
+  return bytes
+}
+
+/**
+ * @notice Write a 64-bit little-endian bigint to a Uint8Array.
+ * @param {bigint} bigint The bigint value to write to the Uint8Array.
+ * @return {Uint8Array} A Uint8Array of length 32 containing the 64-bit little-endian bigint.
+ */
+export function writeBigInt64LE(bigint: bigint): Uint8Array {
+  const bytes = new Uint8Array(32)
+
+  const lo = BigInt.asUintN(32, bigint)
+  const hi = BigInt.asUintN(32, bigint >> BigInt(32))
+
+  bytes[0] = Number(lo & BigInt(0xff))
+  bytes[1] = Number((lo >> BigInt(8)) & BigInt(0xff))
+  bytes[2] = Number((lo >> BigInt(16)) & BigInt(0xff))
+  bytes[3] = Number((lo >> BigInt(24)) & BigInt(0xff))
+
+  bytes[4] = Number(hi & BigInt(0xff))
+  bytes[5] = Number((hi >> BigInt(8)) & BigInt(0xff))
+  bytes[6] = Number((hi >> BigInt(16)) & BigInt(0xff))
+  bytes[7] = Number((hi >> BigInt(24)) & BigInt(0xff))
+
+  return bytes
+}
```

Comment:
These new helper methods look more "advanced" (lol) than our classic conversion helpers for BE! 🙂 🤔 Also love that they explicitly target already 32/64 bit e.g..

Can you say something how they compare and relate to our existing methods, e.g. performance wise? This explicit bit-length also has the nice (and eventually performance-increasing) side effect that it omits the 0-padding if I see correctly?

If these methods are better than our old ones: might it be a mid-term option to also adopt this for our BE methods?

(would like to at least high-level settle on these kind of questions before we take new methods with new naming schemes and everything into Util)

Reply:
I've researched this a bit, and I think we could strike a balance between high performance and readability (the current implementation is quite opaque) by using DataView to go from Uint8Arrays to ints/bigints. I'm reading that DataViews are very performant & optimized, and the methods have built-in support for BE/LE. See this reference for more info: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/DataView/getBigInt64

Reply:
When you mention "adopting this for our BE methods", what methods are you referring to? `bytesToBigInt` and `bytesToInt`? I can implement alternative versions using DataViews and compare performance.